Feature/mstancar ana

eyandel opened the pull request:
https://github.com/SBNSoftware/sbndaq-artdaq-core/pull/83

Commit:
moving SBND specific analysis modules out of common area.  Adding PMT metrics mo
dule

Commit:
Adding PMT metrics

Commit:
Adding PMT metrics

Commit:
Adding PMT metrics

Commit:
right before add read software trigger metrics

Commit:
Lan: add to EventAna to save waveforms from multiple CAEN1730

Commit:
adding default fcl files as examples

Commit:
save changes before aligning with develop

Commit:
Jiaoyang merged some folders

Commit:
fix conflicts

lynnt20 commented:
There is a couple bug fixes I need to copy over (haven't made a PR in sbndcode yet) for the pulse finding function in the PMT metrics module. Could I add those myself and commit? Otherwise looks good!

eyandel commented:
> 

we might want to do those separately so we can check that it compiles in the daq code (since I think you've only tested it in sbndcode, right?). We can talk to the 1730 people to see if they can test it for us first too.

lynnt20 commented:
We can wait to see if it compiles if that makes the most sense, but the changes are really simple (like I think I have to switch a > to a < or something) and are mainly arithmetic.

eyandel commented:
Oh that sounds fine then. You can make the changes here and I can pull it
down and compile it real quick.

On Tue, May 23, 2023 at 12:41 PM lynnt-uchicago ***@***.***>
wrote:

> We can wait to see if it compiles if that makes the most sense, but the
> changes are really simple (like I think I have to switch a > to a < or
> something) and are mainly arithmetic.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/SBNSoftware/sbndaq-artdaq/pull/106#issuecomment-1559881504>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHBWKHV3PEV7DUAQMPAOMUTXHTZEPANCNFSM6AAAAAAYK47XWY>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
--

lynnt20 committed:
fix beam start/end bin definitions, and pulse finding algo 

beam start/end bin definitions: 
- correction: if the trigger timestamp is > 1000, the beginning of the beam spill is **not** contained within the waveform. In this case, the beam start bin is set to 0 (the first bin is already within the beam spill). if the trigger timestamp is < 1000, then the beamStartBin is *not* the first bin, and needs a correction to obtain the bin that corresponds to the start of the beam spill

pulse finding algo
- simplified the definition of where a start/end of a pulse is 
- fixed bug finding the peak of the pulse

lynnt20 commented:
Added the commit! More lengthy description of changes in the commit description

eyandel commented:
> Added the commit! More lengthy description of changes in the commit description

great! I just pulled it down and it compiled fine so we're all good.
